packages/flow/lib/AstroIterate.astro.js

```diff
@@ -1,37 +1,43 @@
-import { getNormalizedGenerator, isIterable } from './shared.js'
+import {
+	getNormalizedGenerator,
+	hasForEachMethod,
+	isIterable,
+} from './shared.js'
 
 export default Object.assign(
-	function Iterate(_result, attributes, slots) {
+	function Iterate(_result, { of: each }, slots) {
 		const promiseOfGenerator = Promise.resolve(
 			typeof slots.default === 'function'
 				? slots.default()
-			: slots.default
-		).then(
-			(result) => result.expressions.at(0)
-		).then(
-			(result) => getNormalizedGenerator(result),
-			() => null
+				: slots.default
 		)
-
-		const iterables = Object.values(attributes)
+			.then((result) => result.expressions.at(0))
+			.then(
+				(result) => getNormalizedGenerator(result),
+				() => null
+			)
 
 		return {
 			[Symbol.toStringTag]: 'AstroComponent',
 
 			async *[Symbol.asyncIterator]() {
 				const generator = await promiseOfGenerator
 
-				for (const iterable of iterables) {
-					if (isIterable(iterable)) {
-						for await (const element of iterable) {
-							yield * generator(element)
+				if (isIterable(each)) {
```

Comment:
This code makes it so that when an iterable with the `forEach()` is passed through that the call back function is passed the keys' values and  the iterable that was sent in